features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListener.kt

```diff
@@ -160,6 +195,42 @@ internal class JankStatsActivityLifecycleListener(
         activeActivities[window] = list
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    private fun registerMetricListener(window: Window, activity: Activity) {
+        if (buildSdkVersionProvider.version() >= Build.VERSION_CODES.S) {
+            if (frameMetricsListener == null) {
+                frameMetricsListener = DDFrameMetricsListener()
+            }
+            val handler = Handler(Looper.getMainLooper())
+            frameMetricsListener?.let { window.addOnFrameMetricsAvailableListener(it, handler) }
+        } else {
+        // Fallback - Android 30 allows apps to not run at a fixed 60hz, but didn't yet have
+        // Frame Metrics callbacks available
+        if (display == null && buildSdkVersionProvider.version() == Build.VERSION_CODES.R) {
+            val displayManager = activity.getSystemService(Context.DISPLAY_SERVICE) as DisplayManager
+            display = displayManager.getDisplay(Display.DEFAULT_DISPLAY)
```

Comment:
I believe holding `Display` object shouldn't cause memory leaks by looking on its structure https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/view/Display.java;l=1?q=Display.java&sq= - it _seems_ it doesn't have references to the UI-related context (Activity), the only concern is `Resources` object which may hold such reference. So maybe it is worth to check this change for the leaks.

Reply:
Didn't see anything on my local profiles. 
Also did a quick analysis on the activity allocation/deallocation pattern in the sample app with and without these changes and found no difference.